### PR TITLE
morpheus 2.3.6 (new formula)

### PR DESCRIPTION
**Pull request by jdieg0**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/m/morpheus.rb
Comment (chenrui333) [inline Formula/m/morpheus.rb:37]:
```suggestion
    args = []

    if OS.mac?
      args << "-DMORPHEUS_RELEASE_BUNDLE=ON"
      args << "-DMORPHEUS_SBML=OFF" # SBML import currently disabled due to libSBML build errors with macOS SDK
    end

    system "cmake", "-S", ".", "-B", "build", *args, *std_cmake_args
    system "cmake", "--build", "build"
    system "cmake", "--install", "build"
```

File: Formula/m/morpheus.rb
Comment (chenrui333) [inline Formula/m/morpheus.rb:21]: any issue of tracking qt6 support?

**Review by chenrui333 — COMMENTED** [review]
Except for a few minor comments, this LGTM

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:21]: Morpheus doesn't support Qt6 yet, so there is unfortunately no other option than Qt5 at the moment, but we're working on that and we'll be adding Qt6 support in an upcoming version.

File: Formula/m/morpheus.rb
Comment (starboerg) [reply Formula/m/morpheus.rb:29]: Latest CMake version started to show issues with tracking dependencies between build targets, which can result in occasional build order issues. The ninja generator did not show this issue. Hopefully that will be resolved in future, but is not in our hands and we kept with the known stable solution "ninja".

Comment (SMillerDev) [reply Formula/m/morpheus.rb:29]: Can we have a reference to the upstream issue here then, so we know when we can use the default system again.

Comment (starboerg) [reply Formula/m/morpheus.rb:29]: Yes, of course ...

https://gitlab.kitware.com/cmake/cmake/-/issues/25142

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:48]: Upstream now only provides the simple placeholder string `HOMEBREW_BIN_PATH` to be replaced in the `brew postinstall`.

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb]: `morpheus-gui` is now created identically to `morpheus` with `bin.write_exec_script`.

**Comment by starboerg** [comment]
Dear @SMillerDev and @chenrui333, please consider reviewing the updated formula

File: Formula/m/morpheus.rb
Comment (SMillerDev) [reply Formula/m/morpheus.rb:29]:
```suggestion
    # has to build with Ninja until: https://gitlab.kitware.com/cmake/cmake/-/issues/25142
    args = []
    args << "-G Ninja"
```

**Review by chenrui333 — APPROVED** [review]
LGTM, thanks!

**Comment by chenrui333** [comment]
```
homebrew/core
  * The following should either be an alias or a rename, not both: bigdata, boot2docker, camlistore, crystal-lang, findbugs, gitlab-ci-multi-runner, gnome-icon-theme, gnupg2, gtef, gtk+4, htop-osx, latexila, letsencrypt, libmongoclient, linux-headers, mat, mobile-shell, mongo-c, newsbeuter, osh, pyqt5, qt5, recipes, speedtest_cli, ssreflect, tachyon, tensorflow and transfig
```

need a rebase before the merge

File: Formula/m/morpheus.rb
Comment (carlocab) [inline Formula/m/morpheus.rb:44]:
```suggestion
    return unless OS.mac?
    
    bin.write_exec_script "#{prefix}/Morpheus.app/Contents/MacOS/morpheus"
    bin.write_exec_script "#{prefix}/Morpheus.app/Contents/MacOS/morpheus-gui"
```

File: Formula/m/morpheus.rb
Comment (carlocab) [inline Formula/m/morpheus.rb:52]: You don't need this in `post_install`; you can move this to the end of the `install` method.

File: Formula/m/morpheus.rb
Comment (carlocab) [inline Formula/m/morpheus.rb:66]:
```suggestion
```
This isn't really Homebrew-specific (for the former), and the latter tends to confuse users (e.g. symlinking app bundles doesn't work with Spotlight)

File: Formula/m/morpheus.rb
Comment (carlocab) [inline Formula/m/morpheus.rb:30]:
```suggestion
    args = ["-G Ninja"]
```

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:44]: Thanks for the suggestion, but we already passed all checks (including style) and according to @chenrui333 we should be ready for merge. I kindly ask that style related matters be modified in the next commit/version bump.

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:52]: The suggestion would potentially break the software as it wouldn't be aware of the target system's `"#{HOMEBREW_PREFIX}/bin"` path, particularly if shipped via bottles, and therefore I would respectfully reject this one.

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:66]: That's a good point. Unfortunately, the app bundle indeed doesn't show up in Spotlight this way. Do you have a suggestion on how this could be achieved?

However, the symlink is important for many users in that the app will appear in the Launchpad.

In addition to the command-line tool `morpheus`, `morpheus-gui` adds important functionality whose existence and usage should be pointed out to the user in the caveats in my opinion. Furthermore, the launch of `morpheus-gui` is relevant for the registration of the URI scheme `morpheus://...`.

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:30]: Same as the style change [above](https://github.com/Homebrew/homebrew-core/pull/162155#discussion_r1562265144). This was added by @SMillerDev in https://github.com/Homebrew/homebrew-core/pull/162155#discussion_r1560576053 and already signed off for merging.

File: Formula/m/morpheus.rb
Comment (SMillerDev) [reply Formula/m/morpheus.rb:44]: Experience teaches me that that postpones it indefinitely. Since this is a suggestion, it should be very easy to incorporate.

File: Formula/m/morpheus.rb
Comment (SMillerDev) [reply Formula/m/morpheus.rb:52]: Why wouldn't it be aware of that?

File: Formula/m/morpheus.rb
Comment (SMillerDev) [reply Formula/m/morpheus.rb:66]: Distributing GUI applications really isn't a goal in Homebrew/core so if this is required it might not be a good fit to add this to Homebrew/core.

File: Formula/m/morpheus.rb
Comment (SMillerDev) [reply Formula/m/morpheus.rb:30]: Because we're all humans and carlocab brings up a good point, this is a nicer way of writing this.

File: Formula/m/morpheus.rb
Comment (carlocab) [reply Formula/m/morpheus.rb:44]: You need to apply https://github.com/Homebrew/homebrew-core/pull/162155#discussion_r1562266473 anyway, so may as well do this now

File: Formula/m/morpheus.rb
Comment (carlocab) [reply Formula/m/morpheus.rb:30]: You need to apply https://github.com/Homebrew/homebrew-core/pull/162155#discussion_r1562266473 anyway, so may as well do this now

File: Formula/m/morpheus.rb
Comment (carlocab) [reply Formula/m/morpheus.rb:52]: Nope, it won't break. `brew` knows to adjust `$HOMEBREW_PREFIX` references in text files when pouring bottles.

File: Formula/m/morpheus.rb
Comment (carlocab) [reply Formula/m/morpheus.rb:66]: > Do you have a suggestion on how this could be achieved?

No, there is no good way for achieving this. That's why we avoid telling users about `.app` bundles installed by formulae, because it just doesn't result in a good user experience. (Users can still find it for themselves, but they're on their own when they do that.)

**Review by carlocab — CHANGES_REQUESTED** [review]
Sorry, I don't think this is ready to merge. But I'm happy to approve this PR when my suggestions are applied (or if another maintainer thinks they are bad suggestions).

File: Formula/m/morpheus.rb
Comment (jdieg0) [reply Formula/m/morpheus.rb:52]: Apologies for having to ask again. 😅

As I understand it, the `inreplace` operation in the `install` block is not executed when a bottle is poured.

The goal here is to communicate non-standard Homebrew prefixes, that may have been configured by the user of the target system, to the application bundle (for this purpose, upstream provides the placeholder string `HOMEBREW_BIN_PATH` in the Info.plist) in order to make runtime dependencies installed via Homebrew findable in the application context as well (see the [Info.plist reference](https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/LaunchServicesKeys.html#//apple_ref/doc/uid/20001431-106825)).

So it would make sense to do this replacement not in the bottling step in the `install` block on the CI system, but after the bottle has been poured during `post_install` on the user's target system.

Unless I'm missing something here?

Comment (carlocab) [reply Formula/m/morpheus.rb:51]: You are.

As I've explained: `brew` knows to adjust references to `$HOMEBREW_PREFIX` inside text files on the build machine so that they are appropriate for the machine that the bottle is being poured in (which may have a different `HOMEBREW_PREFIX`).

You don't need to manually manage this inside a `post_install` block because `brew` already knows to fix it up for you.

File: Formula/m/morpheus.rb
Comment (carlocab) [inline Formula/m/morpheus.rb:48]:
```suggestion
    # Set PATH environment variable including Homebrew prefix in macOS app bundle
    inreplace "#{prefix}/Morpheus.app/Contents/Info.plist", "HOMEBREW_BIN_PATH", "#{HOMEBREW_PREFIX}/bin"
```
Let's just do this unconditionally.

**Review by carlocab — APPROVED** [review]
LGTM after one last change. Thanks for your patience!

**Comment by jdieg0** [comment]
@carlocab Final change is now added (https://github.com/Homebrew/homebrew-core/pull/162155/commits/dd2c55940edfc6e9f9bcdd5aa15e25aa215f3da8).

Thanks to everyone!

**Review by carlocab — APPROVED** [review]

**Pull request merged**
